src/java.base/share/classes/java/io/Reader.java

```diff
@@ -382,6 +396,111 @@ public int read(char[] cbuf) throws IOException {
      */
     public abstract int read(char[] cbuf, int off, int len) throws IOException;
 
+    private String readAllCharsAsString() throws IOException {
+        StringBuilder result = new StringBuilder();
+        char[] str = new char[TRANSFER_BUFFER_SIZE];
+        int n;
+        while ((n = read(str)) != -1) {
+            result.append(str, 0, n);
+        }
+        return result.toString();
+    }
+
+    /**
+     * Reads all remaining characters as lines of text. This method blocks until
+     * all remaining characters have been read and end of stream is detected,
+     * or an exception is thrown. This method does not close the reader.
+     *
+     * <p> When this reader reaches the end of the stream, further
+     * invocations of this method will return an empty list.
+     *
+     * <p> A <i>line</i> is either a sequence of zero or more characters
+     * followed by a line terminator, or it is a sequence of one or
+     * more characters followed by the end of the stream.
+     * A line does not include the line terminator.
+     *
+     * <p> A <i>line terminator</i> is one of the following:
+     * a line feed character {@code "\n"} (U+000A),
+     * a carriage return character {@code "\r"} (U+000D),
+     * or a carriage return followed immediately by a line feed
+     * {@code "\r\n"} (U+000D U+000A).
+     *
+     * <p> The behavior for the case where the reader is
+     * <i>asynchronously closed</i>, or the thread interrupted during the
+     * read, is highly input stream specific, and therefore not specified.
```

Comment:
The carry over of this paragraph needs to replace "input stream" with reader (same thing in the other method).

Reply:
Yeah, I slipped up bringing the words over from `InputStream`.

Reply:
Fixed in 98387b0.